sessionctx/variable/sysvar.go

```diff
@@ -125,6 +125,10 @@ var defaultSysVars = []*SysVar{
 		s.enforceMPPExecution = TiDBOptOn(val)
 		return nil
 	}},
+	{Scope: ScopeSession, Name: TiFlashMaxThreads, Type: TypeInt, Value: strconv.Itoa(DefTiFlashMaxThreads), MinValue: -1, MaxValue: 256, SetSession: func(s *SessionVars, val string) error {
+		s.TiflashMaxThreads = TidbOptInt64(val, DefTiFlashMaxThreads)
+		return nil
+	}},
```

Comment:
```suggestion
	{Scope: ScopeGlobal | ScopeSession, Name: TiFlashMaxThreads, Type: TypeInt, Value: strconv.Itoa(DefTiFlashMaxThreads), MinValue: -1, MaxValue: MaxConfigurableConcurrency, SetSession: func(s *SessionVars, val string) error {
		s.TiflashMaxThreads = TidbOptInt64(val, DefTiFlashMaxThreads)
		return nil
	}},
```

One other question - what happens when the user sets the concurrency to zero? If this is a problem you can change the definition to:

```
MinValue: 1, AllowAutoValue: true
```

(AutoValue is a special value of -1. In MySQL configuration terms it means "set it for me").

Reply:
MaxValue changed.

> One other question - what happens when the user sets the concurrency to zero? If this is a problem you can change the definition to:
> 
> ```
> MinValue: 1, AllowAutoValue: true
> ```
> 
> (AutoValue is a special value of -1. In MySQL configuration terms it means "set it for me").

That max_threads = 0 has special means.
set max_threads =
-1 : TiFlash will use its own max_threads config;
0 : max_threads will be pushed down to TiFlash and it means that the concurrency will determined automatically;
x(>0) : max_threads will be pushed down to TiFlash and it means that the concurrency will be set to x.